### ISSUE TITLE
Autocorrecting Style/LineEndConcatenation changes behaviour

### ISSUE THREAD

**Issue by Bombasarkadian** [rubocop-hq/rubocop#6597]
When using `<<` to append multiple string to array, `Style/LineEndConcatenation` suggests using `\` to join the strings together.

```ruby
array = []
array << 'Foo' <<
         'Bar' <<
         'Baz'
puts array.join('-')
```

---

## Expected behavior

The way we append strings to array should not be changed. Running corrected code should output `Foo-Bar-Baz`

## Actual behavior

Autocorrect changes the code to use `\` instead of `<<` which causes strings to be joined before appending them to the array. Running corrected code outputs `FooBarBaz`

## Steps to reproduce the problem

Autocorrect the code snippet above with `rubocop -a`

## RuboCop version

```
$ rubocop -V
0.61.1 (using Parser 2.5.3.0, running on ruby 2.5.3 x86_64-linux)
```

**Comment by Drenmi** [comment]
Unfortunately we can't know what type a variable dereferences to at runtime using static analysis. What we can do is mark this auto-correction as unsafe in the configuration.

**Comment by jaredbeck** [comment]
Fixed by https://github.com/rubocop-hq/rubocop/pull/6605
